### ISSUE TITLE
is it possible to add "tapinstall.exe" to this repo ?

### ISSUE THREAD

**Issue by chipitsine** [OpenVPN/tap-windows6#35]
I want to reproduce tap-windows6 build in AppVeyor, "tapinstall.exe" is missing.
also, there's no clear guide where am I supposed to download it.

**Comment by mattock** [comment]
Tapinstall.exe is actually called devcon.exe. The source code is available, but it has - or used to have - pretty severe limitations on redistribution. Also, building it can be tricky with recent Visual Studio / C compiler versions. Right now I'm using an old tapinstall.exe build by @jamesyonan.

That said, Microsoft has been actively pushing its code into GitHub. So it is possible that devcon.exe sources have been open sourced as well. If that is the case, we could fix devcon.exe sources so that they compile cleanly and even include them in the tap-windows6 repository.

**Comment by chipitsine** [comment]
Yes, I saw devcon sources on github, what kind of modification should be made to its source?

**Comment by mattock** [comment]
Where are the sources exactly?

**Comment by chipitsine** [comment]
https://github.com/Microsoft/Windows-driver-samples/tree/master/setup/devcon

**Comment by chipitsine** [comment]
it is not hard to build it in AppVeyor.  it is pretty recent, so, it builds fine

**Comment by mattock** [comment]
Thanks! The license is MS-PL which is an [OSI-approved license](https://opensource.org/licenses/MS-PL). I think we should fork https://github.com/Microsoft/Windows-driver-samples/ under the OpenVPN organization and move on from there. Two ideas:

1. Build tapinstall.exe from sources in buildtap.py
1. Build static 32-bit and 64-bit tapinstall.exe manually and place the resulting binaries to the tap-windows6 repository.

Thoughts?

**Comment by chipitsine** [comment]
https://ci.appveyor.com/project/chipitsine/windows-driver-samples/build/artifacts

**Comment by chipitsine** [comment]
what modification is required to the source?

**Comment by chipitsine** [comment]
well, we can add windows-driver-sample as a submodule to tap-windows6.
and as for buildtap.py ... brrrr.... I wish to convert to Powershell/MSBuild and combine with https://github.com/mattock/sign-tap6

**Comment by mattock** [comment]
> what modification is required to the source?

Can't recall. If you managed to build it, then my problem was just some incompatibility between my Visual Studio and devcon source versions.

> well, we can add windows-driver-sample as a submodule to tap-windows6.

Indeed. If we don't have to modify the devcon sources at all, then that would be sufficient.

**Comment by mattock** [comment]
Simon on openvpn-devel has a tapinstall.exe replacement. And I believe @jkunkee mentioned that devcon.exe sources are now available under MS-PL license which means we could easily bundle those. Or add those as a Git submodule.

**Comment by jkunkee** [comment]
Devcon is indeed now MS-PL and up on Github as @mattock described earlier in this thread. :)

@chipitsine, the only change I had to make was renaming the binary output from devcon.exe to tapinstall.exe. The current driver install process uses stock devcon.exe functionality.

#53 makes it so no changes are needed to devcon's sources.
